### PR TITLE
Chore: Check PR Title on every submission

### PR DESCRIPTION
**Pull request by KevLehman**
## Further comments

We just need to define the list of Prefixes to use, and if we will do any other validation (the action supports using regexes to validate PR titles, but, i don't think we need it 😬)

### CODE REVIEW

File: .github/pr-title-checker-config.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+	"LABEL": {
+	  "name": "Title needs format :)",
+	  "color": "EEEEEE"
+	},
+	"CHECKS": {
+	  "prefixes": ["Bump", "Fix:", "Improve:", "Regression:", "Patch:", "Hotfix:", "[FIX]", "[IMPROVE]", "[REGRESSION]", "[PATCH]", "[HOTFIX]", ["Snyk"]],
```
Comment (ggazzo) [inline .github/pr-title-checker-config.json:7]:
```suggestion
	  "regexp": "^(((\[(NEW|BREAK|IMPROVE|FIX)\](\[(ENTERPRISE|APPS)\])? )|((Regression|Chore|Bump): )).+|(Release [0-9]+\.[0-9]+.[0-9]+))",
```

**Review by ggazzo — DISMISSED** [review]

File: .github/pr-title-checker-config.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+	"LABEL": {
+	  "name": "Title needs format :)",
+	  "color": "EEEEEE"
+	},
+	"CHECKS": {
+	  "regexp": "^(((\[(NEW|BREAK|IMPROVE|FIX)\](\[(ENTERPRISE|APPS)\])? )|((Regression|Chore|Bump): )).+|(Release [0-9]+\.[0-9]+.[0-9]+))",
```
Comment (ggazzo) [inline .github/pr-title-checker-config.json:7]:
```suggestion
	  "regexp": "^(((\\[(NEW|BREAK|IMPROVE|FIX)\\](\\[(ENTERPRISE|APPS)\\])? )|((Regression|Chore|Bump): )).+|(Release [0-9]+\\.[0-9]+.[0-9]+))",
```

**Comment by sampaiodiego** [comment]
@pierre-lehnen-rc might want to give his opinion here as we have been discussing not using `Regression` anymore, but instead chose a new word to describe what a regression means to us.

**Comment by ggazzo** [comment]
I think we have been using this definition for a long time and modifying it in action it is easy, the discussion to find another term can take some time and generate changes in the documentation, in the meantime I think we should already be using validation

**Comment by KevLehman** [comment]
I agree with Gazzo on that. We can defer the discussion about the new term, and start validating the new PRs.

**Review by ggazzo — DISMISSED** [review]

File: .github/pr-title-checker-config.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+	"LABEL": {
+	  "name": "Title needs format :)",
+	  "color": "EEEEEE"
+	},
+	"CHECKS": {
+	  "regexp": "^(((\\[(NEW|BREAK|IMPROVE|FIX)\\](\\[(ENTERPRISE|APPS)\\])? )|((Regression|Chore|Bump): )).+|(Release [0-9]+\\.[0-9]+.[0-9]+))",
```
Comment (sampaiodiego) [inline .github/pr-title-checker-config.json:7]: what if we use "prefixes" to validate the most common usages, so it is easier to read and maintain, and use "regexp" to validate only edge cases like the second group and the "release 1.2.3"
```suggestion
	"CHECKS": {
		"prefixes": ["[NEW]", "[FIX]", "[IMPROVE]", "[BREAK]", "Regression:", "Chore:", "Bump", "Merge master into develop & Set version"]
		"regexp": "(^\\[[^]+]\\[(ENTERPRISE|APPS)\\]|Release [0-9]+\\.[0-9]+\\.[0-9]+)",```

**Review by sampaiodiego — CHANGES_REQUESTED** [review]
Even though my suggestion is not accepted, there is an issue with the provided regexp, that's why the request changes

File: .github/pr-title-checker-config.json (hunk 3 above)
Comment (sampaiodiego) [reply .github/pr-title-checker-config.json:7]: I don't like Snyk PRs :thinking: we might need to fix it's PR titles. fixed the code suggestion.. but feel free to not use it too =)

Comment (sampaiodiego) [reply .github/pr-title-checker-config.json:7]: the only reason to support Snyk PRs as they are is that if we want to automate merging that kind of version bumps.

Comment (ggazzo) [reply .github/pr-title-checker-config.json:7]: I put the expressions to force the user to insert the spaces and ":" using prefix its not possible =/.
I dont think the ability to read needs to be over the capacity to express the rules: it should cover your specification and that should be documented. if the expression to cover that is hard 🤷 there is nothing that we can do about it

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**